src/main/java/com/onarandombox/MultiverseSignPortals/listeners/MVSPPlayerListener.java

```diff
@@ -83,28 +86,51 @@ public void playerPortal(PlayerPortalEvent event) {
      */
     @EventHandler
     public void playerInteract(PlayerInteractEvent event) {
+
+
+        // The event must not be canceled...
         if (event.isCancelled()) {
             return;
         }
-        if (event.getAction() == Action.RIGHT_CLICK_BLOCK) {
-            if (event.getClickedBlock().getState() instanceof Sign) {
-                Logging.finer("Found a Sign!");
-                Sign s = (Sign) event.getClickedBlock().getState();
-                SignStatus status = this.pd.getSignStatus(s);
-                if (status == SignStatus.SignPortal) {
+
+        // We must be right-clicking...
+        if (event.getAction() != Action.RIGHT_CLICK_BLOCK) {
+            return;
+        }
+
+        // And it must be a sign
+        if (!(event.getClickedBlock().getState() instanceof Sign)) {
+            return;
+        }
+
+        Logging.finer("Found a Sign!");
+        Sign s = (Sign) event.getClickedBlock().getState();
+        SignStatus status = this.pd.getSignStatus(s);
+
+        Player player = event.getPlayer();
+        switch (status) {
+            case SignPortal:
+                if (permissions.hasPermission(player, USE_PERMISSION, false)) {
                     String destString = this.pd.processSign(s);
-                    this.takePlayerToDestination(event.getPlayer(), destString);
-                    event.setCancelled(true);
-                } else if (status == SignStatus.Disabled) {
-                    this.pd.activateSignPortal(event.getPlayer(), ChatColor.RED + "Disabled", s);
-                    event.setCancelled(true);
-                } else if (status == SignStatus.Legacy) {
-                    this.pd.activateSignPortal(event.getPlayer(), ChatColor.AQUA + "Legacy", s);
-                    event.setCancelled(true);
-                } else if (status == SignStatus.NetherPortalSign) {
-                    event.setCancelled(true);
+                    this.takePlayerToDestination(player, destString);
+                } else {
+                    player.sendMessage(ChatColor.RED + "You do not have the required permission to use SignPortals (" + USE_PERMISSION + ")");
                 }
-            }
+                event.setCancelled(true);
+                break;
+            case Legacy:
+                this.pd.activateSignPortal(player, ChatColor.AQUA + "Legacy", s);
+                event.setCancelled(true);
+                break;
+            case Disabled:
+                this.pd.activateSignPortal(player, ChatColor.RED + "Disabled", s);
+                event.setCancelled(true);
+                break;
+            case NetherPortalSign:
+                event.setCancelled(true);
+                break;
+            case NotASignPortal: // We shouldn't bother with other people's stuff
+                event.setCancelled(false);
```

Comment:
This added case really doesn't need to exist. The event should already not be cancelled so there's really no reason to set it to not cancelled again.

Reply:
Ah, I can remove it, added it there to fill out the Enum mainly. I guess what little Rust I’ve done has affected me 😆